packages/website/src/components/ASTViewerScope.tsx

```diff
@@ -17,12 +17,8 @@ export default function ASTViewerScope({
   const [model, setModel] = useState<string | ASTViewerModelMap>('');
 
   useEffect(() => {
-    if (typeof value === 'string') {
-      setModel(value);
-    } else {
-      const scopeSerializer = createScopeSerializer();
-      setModel(serialize(value, scopeSerializer));
-    }
+    const scopeSerializer = createScopeSerializer();
+    setModel(serialize(value, scopeSerializer));
   }, [value]);
```

Comment:
We can go even simpler here and in the other two `ASTViewer(.+)` components:

```tsx
const astSerializer = createESTreeSerializer();

export default function ASTViewerESTree({
  value,
  position,
  onSelectNode,
}: ASTESTreeViewerProps): JSX.Element {
  const model = serialize(value, astSerializer);

  return (
    <ASTViewer value={model} position={position} onSelectNode={onSelectNode} />
  );
}
```

This is similar to the other thread about `useEffect` _(I'm forgetting where that was...)_: it's not generally needed to synchronize some data as the result of some inputs. You can use a variable directly instead, and the computed data (`model`) will reflect whatever the latest prop data (`value`) is at each render.

If performance is a concern you can use a `useMemo`:

```ts
const model = useMemo(() => serialize(value, astSerializer), [value]);
```